working-release-notes/README.md

```diff
@@ -0,0 +1,45 @@
+# Bevy's Release Notes Process
+
+Hi! Did someone add `M-Needs-Release-Note` to your PR? If so, you're in the right place.
+Let's talk about how this process works.
+
+When we make high-impact changes to Bevy, we need to communicate them to users (and potential users!).
+For the most part, this is done via our famously in-depth release notes.
```

Comment:
It'd be nice to link to these famous release notes somewhere to show an example of what they *actually* look like. Maybe just a link to <https://bevyengine.org/news/>, or to some specific release as an example?

(Side note: If we supported filtering news posts on the Bevy website, we could do e.g. <https://bevyengine.org/news?topics=release> to directly link to a view of all release notes)